grace/models/feature_extractor.py

```diff
@@ -0,0 +1,124 @@
+from typing import Any, Dict, Tuple, Callable
+
+import torch
+
+from torchvision.models import resnet152, ResNet152_Weights
+from torchvision.transforms import (
+    Resize,
+    Lambda,
+    Normalize,
+    Compose,
+    RandomApply,
+)
+
+from grace.base import GraphAttrs
+from grace.utils.augment_image import RandomEdgeCrop
+
+
+def classifier_to_feature_extractor(classifier: torch.nn.Module) -> torch.nn.Module:
+    """Takes a pre-trained classification model and returns a copy in which
+    the final (classification) layer has been removed.
+    
+    Parameters
+    ----------
+    classifier : torch.nn.Module
+        Classifier model
+
+    Returns
+    -------
+    extractor : torch.nn.Module
+        Feature extractor model
+    """
+    modules = list(classifier.children())[:-1]
+    extractor = torch.nn.Sequential(*modules)
+    for p in extractor.parameters():
+        p.requires_grad = False
+
+    return extractor
+
+
+default_transforms = Compose(
+    [
+        Resize(size=(224, 224)),
+        Normalize(mean=[0.0], std=[1.0]),
+        Lambda(lambda x: x.expand(1, 3, 224, 224)),
+    ]
+)
+default_augmentations = RandomApply([RandomEdgeCrop(max_fraction=0.1)], p=0.3)
+
+
+class FeatureExtractor(torch.nn.Module):
+    """A feature extraction model. Crops bounding boxes according to
+    graph coordinates then calculates features with a specified model.
+
+    Parameters
+    ----------
+    bbox_size : Tuple[int]
+        Size the bounding boxes to be extracted, centered
+        on the x-y coordinates of each node; (W, H)
+    model : Callable
+        Feature extractor model or function that maps a tensor
+        to a tensor
+    transforms : Callable
+        Series of transforms to apply to the bbox image before
+        feature extraction
+    augmentations : Callable
+        Series of augmentations to apply to the bbox image
+        during training
+    """
+
+    def __init__(
+        self,
+        *,
+        bbox_size: Tuple[int] = (224, 224),
+        model: Callable = classifier_to_feature_extractor(resnet152(ResNet152_Weights)),
+        transforms: Callable = default_transforms,
+        augmentations: Callable = default_augmentations,
```

Comment:
I suggest you make this a non-optional argument that needs to be added when you set this up

```suggestion
    def __init__(
        self,
        model: Callable,
        *,
        bbox_size: Tuple[int] = (224, 224),
        transforms: Callable = default_transforms,
        augmentations: Callable = default_augmentations,
```